crud/src/main/java/org/openmbee/sdvc/crud/services/DefaultNodeService.java

```diff
@@ -81,12 +82,22 @@ public void setNodeGetHelper(NodeGetHelper nodeGetHelper) {
         this.nodeGetHelper = nodeGetHelper;
     }
 
+    @Autowired
+    public void setEventPublisher(Optional<EventPublisher> eventPublisher) {
+        this.eventPublisher = eventPublisher;
+    }
+
     @Override
     public ElementsResponse read(String projectId, String refId, String id,
         Map<String, String> params) {
 
+        EventObject event = EventObject.create(projectId, refId, "READ", params);
```

Comment:
remove

Reply:
also needs to add event publishing for commit, ref creation and project creation, i think we can create subclasses of eventobject for CommitEvent, ProjectEvent, RefEvent so listeners have more flexibility on which to listen to